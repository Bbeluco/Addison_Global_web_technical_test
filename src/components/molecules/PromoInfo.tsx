import React from 'react';
import {TextInformationAboutPromo} from '../atoms/TextInformationAboutPromo';
import {StyleSheet, View} from 'react-native';
import {TitleNamePromo} from '../atoms/TitleNamePromo';
import {stylesType} from '../../types/stylesType';

export function PromoInfo({
  styleParent,
}: Omit<stylesType, 'styleChildren'>): React.JSX.Element {
  return (
    <View style={styleParent}>
      <TitleNamePromo text="TITULO" styleChildren={styleTitleNamePromo.text} />
      <TextInformationAboutPromo
        text="Lorem ipsum gigante"
        styleChildren={styleTextInformationAboutPromo.text}
      />
    </View>
  );
}

const styleTextInformationAboutPromo = StyleSheet.create({
  text: {
    fontSize: 12,
    color: 'gray',
    fontWeight: '600',
  },
});

const styleTitleNamePromo = StyleSheet.create({
  text: {
    fontSize: 24,
  },
});
