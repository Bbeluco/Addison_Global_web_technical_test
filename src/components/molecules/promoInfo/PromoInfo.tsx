import React from 'react';
import {TextInformationAboutPromo} from '../../atoms/textInformationAboutPromo/TextInformationAboutPromo';
import {StyleSheet, View} from 'react-native';
import {TitleNamePromo} from '../../atoms/titleNamePromo/TitleNamePromo';
import {PromoInfoType} from '../../../types/promoInfoType';

export function PromoInfo({
  styleParent,
  titlePromo,
  descPromo,
}: PromoInfoType): React.JSX.Element {
  return (
    <View style={styleParent}>
      <TitleNamePromo
        text={titlePromo}
        styleChildren={styleTitleNamePromo.text}
      />
      <TextInformationAboutPromo
        text={descPromo}
        styleChildren={styleTextInformationAboutPromo.text}
      />
    </View>
  );
}

const styleTextInformationAboutPromo = StyleSheet.create({
  text: {
    fontSize: 12,
    color: 'gray',
    fontWeight: '600',
    padding: 10,
  },
});

const styleTitleNamePromo = StyleSheet.create({
  text: {
    fontSize: 24,
  },
});
