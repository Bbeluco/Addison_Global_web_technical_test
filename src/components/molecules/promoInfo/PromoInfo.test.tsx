import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {PromoInfo} from './PromoInfo';

describe('<PromoInfo />', () => {
  it('UI test', () => {
    const titlePromo = 'New promo';
    const descPromo = 'ABC123';
    render(
      <PromoInfo
        styleParent={{}}
        titlePromo={titlePromo}
        descPromo={descPromo}
      />,
    );

    expect(screen.queryByText(titlePromo)).not.toBeNull();
    expect(screen.queryByText(descPromo)).not.toBeNull();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
