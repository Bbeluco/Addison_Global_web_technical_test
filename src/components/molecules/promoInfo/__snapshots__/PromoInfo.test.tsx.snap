// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoInfo /> UI test 1`] = `
<View
  style={{}}
>
  <View>
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      New promo
    </Text>
  </View>
  <View>
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 12,
          "fontWeight": "600",
          "padding": 10,
        }
      }
    >
      ABC123
    </Text>
  </View>
</View>
`;
